modules/getting-started/assets/common/quickstart-java27.adoc

```diff
@@ -0,0 +1,283 @@
+== Step 2: Adding Couchbase Dependency
+
+
+The first step is to add following block above the tag called *</project>* near the end of your *pom.xml* file:
+
+[source,XML]
+----
+<dependencies>
+  <dependency>
+    <groupId>com.couchbase.client</groupId>
+    <artifactId>java-client</artifactId>
+    <version>2.7.14</version>
+  </dependency>
+</dependencies>
+
+----
+
+
+== Step 3: Connecting to Couchbase
+
+We will build a simple line command app. Let's first show how to connect with Couchbase:
+
+
+[source,Java]
+----
+
+
+import com.couchbase.client.java.*;
+import com.couchbase.client.java.document.*;
+import com.couchbase.client.java.document.json.*;
+import com.couchbase.client.java.query.*;
+
+public class QuickStartCouchbase {
+
+    public static void main(String[] args) {
+
+        String bucketName = "default";
+        String username = "Administrator";
+        String password = "password";
+        Cluster cluster = null;
+        Bucket bucket = null;
+
+        try {
+
+            // Initialize the Connection
+            cluster = CouchbaseCluster.create("localhost");
+            cluster.authenticate(username, password);
+            bucket = cluster.openBucket(bucketName);
+            
+            
+        } catch (Exception e ) {
+            System.out.println("Ooops!... something went wrong");
+            e.printStackTrace();
+        } finally {
+
+            //Naive way to just close a single bucket
+            if(bucket != null) {
+                bucket.close();
+            }
+
+            // Disconnect and close all buckets
+            if(cluster != null) {
+                cluster.disconnect();
+            }
+        }
+
+    }
+}
+
+
+----
+
+IMPORTANT: It is very important that cluster and bucket instances are created during startup and are then reused until the application shuts down. Connection setup is expensive and the SDK is designed to be thread safe and can be efficiently used across all your application threads.
+
+
+The cluster and bucket objects can be reused in your application, the most common pattern is to create a link:https://en.wikipedia.org/wiki/Singleton_pattern[Singleton] returning them. You can read more about link:https://docs.couchbase.com/java-sdk/2.7/managing-connections.html[how to manage your connection here].
+
+== Step 4: Create and query new document
+
+
+Suppose we need to store the following data in the database representing 3 distinct users:
+
+[source,JavaScript]
+----
+//user1
```

Comment:
aren't these keys supposed to be of the form "user::0001"?

Reply:
Done! thanks